File: code/Java/observer/weather/WeatherData.java
package observer.weather;

import observer.observer.Observer;
import observer.subject.Subject;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class WeatherData implements Subject {
    private List<Observer> observers;   // other data structure
    private float temperature;
    private float humidity;
    private float pressure;

    public WeatherData() {
        observers = new ArrayList<>();
    }

    @Override
    public void registerObserver(Observer observer) {
        observers.add(observer);
    }

    @Override
    public void removeObserver(Observer observer) {
        int index = observers.indexOf(observer);
        if(index >= 0)
            observers.remove(index);
    }

    @Override
    public void notifyObserver() {
        for(Observer observer : observers) {
            observer.update();
        }
    }

    public void measurementsChanged() {
        notifyObserver();
    }

    public void setMeasurements(Object... objects) {
        this.temperature = (float)objects[0];
        this.humidity = (float)objects[1];
        this.pressure = (float)objects[2];
        measurementsChanged();
    }

    public float getTemperature() {
        return temperature;
    }

    public float getHumidity() {
        return humidity;
    }

    public float getPressure() {
        return pressure;
    }
}

File: code/Python/observer/weather/weatherdata.py
from subject.subject import Subject
from observer.observer import Observer

class WeatherData(Subject):
    def __init__(self) -> None:
        self.__observers:list[Observer] = [] 
        self.__temperature = 0
        self.__humidity = 0
        self.__pressure = 0        
    
    def registerObserver(self, o:Observer) -> None:
        self.__observers.append(o)
    
    def removeObserver(self, o:Observer) -> None:
        self.__observers.remove(o)
    
    def notifyObserver(self) -> None:
        for observer in self.__observers:
            observer.update()
    
    def measurementsChanged(self) -> None:
        self.notifyObserver()

    def setMeasurements(self, t:float, h:float, p:float) -> None:
        self.__temperature = t
        self.__humidity = h
        self.__pressure = p
        self.measurementsChanged()
    
    def getTemperature(self) -> float:
        return self.__temperature
    
    def getHumidity(self) -> float:
        return self.__humidity
    
    def getPressure(self) -> float:
        return self.__pressure